controller/src/main/java/org/jboss/as/controller/logging/ControllerLogger.java

```diff
@@ -3629,4 +3629,10 @@ OperationFailedRuntimeException capabilityAlreadyRegisteredInContext(String capa
 
     @Message(id = 476, value = "Value for attribute '%s' is invalid: either '%s' must be specified on its own or '%s' needs to be specified with at least one of '%s' or '%s'")
     OperationFailedException invalidCredentialReferenceValue(String attributeName, String clearTextAttributeName, String credentialStoreAttributeName, String clearTextAttributeName1, String alias);
+
+    @Message(id = 478, value = "Value '%s' is not a legal charset name")
+    String illegalCharsetName(String charset);
```

Comment:
This could be changed from `String` to `OperationFailedException`. Same below. Then, you could simplify the code in your validator to just `throw ControllerLogger.ROOT_LOGGER.illegalCharsetName(...)`.